src/3d/shaders/pointcloud.vert

```diff
@@ -4,16 +4,20 @@ uniform mat4 modelViewProjection;
 
 uniform mat4 projectionMatrix;
 uniform mat4 viewportMatrix;
+uniform mat4 modelMatrix;
 uniform float u_pointSize;
 // used parameter to choose point cloud points color: 0 for height, 1 for classID
 uniform int u_renderingParameter;
 
 in vec3 vertexPosition;
 in float vertexParameter;
 in vec3 vertexColor;
+in vec3 vertexNormal;
 
 out float parameter;
 out vec3 pointColor;
+out vec3 worldPosition;
+out vec3 vertNorm;
```

Comment:
it is worth adding a comment that vertexNormal + worldPosition + vertNorm are only used when doing triangulation.

By the way I am a bit surprised that OpenGL does not complain if there is vertexNormal input when doing normal point cloud rendering, but the actual input attribute is not defined :)

Reply:
If I remenber well (not sure..), I think openGL do not care of any stuff that is not used. So if we have a direction that leads to not use some declaration at any moment, no need to take care of this declaration.